packages/graphql-plugin-auth/src/JWTPlugin.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import jsonwebtoken from "jsonwebtoken";
+import Debug from "debug";
+import { DEBUG_PREFIX } from "./constants";
+
+const debug = Debug(DEBUG_PREFIX);
+
+export interface JWTPluginInput {
+    secret: string;
+    noVerify?: boolean;
+    rolesPath?: string;
+}
+
+class JWTPlugin {
+    secret: string;
+    noVerify?: boolean;
+    rolesPath?: string;
+
+    constructor(input: JWTPluginInput) {
+        this.secret = input.secret;
+        this.noVerify = input.noVerify;
+        this.rolesPath = input.rolesPath;
+    }
+
+    /* eslint-disable @typescript-eslint/require-await */
```

Comment:
**question:** I'm assuming this is `async` because the "contract" is, and the JWKS plugin needs to be? Could we possible explicitly return a `Promise` to get around this ESLint issue, even though nothing is asynchronous?

Reply:
mmm, I think using `return Promise.resolve(result)` is better than disabling the rule tbh, but happy to keep it this way